hub-js/src/local/mutation/create-type-instance.ts

```diff
@@ -0,0 +1,31 @@
+import { Context } from "./context";
+import {
+  CreateTypeInstanceInput,
+  CreateTypeInstancesInput,
+  TypeInstanceUsesRelationInput,
+} from "../types/type-instance";
+import { createTypeInstances } from "./create-type-instances";
+
+interface createTypeInstancesArgs {
+  in: CreateTypeInstanceInput;
+}
+
+export async function createTypeInstance(
+  obj: any,
```

Comment:
1. This mutation returns quite confusing error:

```graphql
mutation CreateTypeInstances {
  createTypeInstance(
    in:  {
        alias: "helm-release"
        typeRef: { path: "cap.type.helm.chart.release", revision: "0.1.0" }
        value: {
          key: "test" # same as it was
        }
        backend: {
          id: "123" # it's like that already.
          context: { # new property of type `Any!`
            name: "release-name",
            namespace: "release-namespace",
          }
        }
        attributes: [{ path: "cap.attribute.sample", revision: "0.1.0" }]
      }
  ) {
    id
    alias
  }
}
```

Result:
```
{
  "errors": [
    {
      "message": "failed to create TypeInstance: failed to create the TypeInstances: Failed to create some TypeInstances. Please verify, if you provided all the required fields for TypeInstances.",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "createTypeInstance"
      ],
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR",
        "exception": {
          "stacktrace": [
            "Error: failed to create TypeInstance: failed to create the TypeInstances: Failed to create some TypeInstances. Please verify, if you provided all the required fields for TypeInstances.",
            "    at /Users/pkosiec/repositories/capact/hub-js/dist/src/local/mutation/create-type-instance.js:28:19",
            "    at Generator.throw (<anonymous>)",
            "    at rejected (/Users/pkosiec/repositories/capact/hub-js/dist/src/local/mutation/create-type-instance.js:6:65)",
            "    at processTicksAndRejections (node:internal/process/task_queues:96:5)"
          ]
        }
      }
    }
  ],
  "data": null
}
```

Same with multiple TIs:

```graphql
mutation CreateTypeInstances {
  createTypeInstances(
    in: {
     typeInstances: [
      {
        alias: "helm-release"
        typeRef: { path: "cap.type.helm.chart.release", revision: "0.1.0" }
        value: {
          key: "test" # same as it was
        }
        backend: {
          id: "123" # it's like that already.
          context: { # new property of type `Any!`
            name: "release-name",
            namespace: "release-namespace",
          }
        }
      }    
    ]
    usesRelations: []
  }
  ) {
    id
    alias
  }
}
```

We should return a meaningful error that a given Backend doesn't exist, because that is the case here I think.

2. Interestingly enough, when I do such mutation:
```graphql
mutation CreateTypeInstances {
  createTypeInstance(
    in:  {
        alias: "helm-release"
        typeRef: { path: "cap.type.helm.chart.release", revision: "0.1.0" }
        value: {
          key: "test" # same as it was
        }
        backend: {
          id: "318b99bd-9b26-4bc1-8259-0a7ff5dae61c" # builtin storage ID
          context: { # new property of type `Any!`
            name: "release-name",
            namespace: "release-namespace",
          }
        }
        attributes: [{ path: "cap.attribute.sample", revision: "0.1.0" }]
      }
  ) {
    id
    alias
  }
}
```

I don't get a response. Same when I remove the `backend.context` from the input. When I delete whole `backend` from input, it works fine.
I assume that we treat any `id` provided in `backend` as some external storage backend, correct? But if that is the case, it is strange we don't get error that a given TI doesn't exist or something like that.

3. I think we would need some additional logging on the Hub side cause it's hard to see what's happening in the cases above

Reply:
all addressed 👍